deployer/src/infra/hanodeagent.rs

```diff
@@ -0,0 +1,20 @@
+use super::*;
+
+#[async_trait]
+impl ComponentAction for HANodeAgent {
+    fn configure(&self, options: &StartOptions, cfg: Builder) -> Result<Builder, Error> {
+        Ok(if !options.ha_node_agent {
+            cfg
+        } else {
+            let spec = ContainerSpec::from_binary("ha-node", Binary::from_dbg("ha-node"));
+            cfg.add_container_spec(spec)
+        })
+    }
+
+    async fn start(&self, options: &StartOptions, cfg: &ComposeTest) -> Result<(), Error> {
+        if options.ha_node_agent {
+            cfg.start("ha-node").await?;
+        }
+        Ok(())
+    }
+}
```

Comment:
Any reason for not implementing `wait_on` ?

Reply:
for now, node-agent is not having any server. So skipped it. Will add with unimplemented.

Reply:
This was not what I meant with unimplemented, this was cause issues as it will panic if you start deployer with wait. I meant adding a:
`rpc Probe (ProbeRequest) returns (ProbeResponse) {}`
to the proto and for now we could simply add unimplemented on the probe impl of the trait.